### PR TITLE
remoteuser - Patch for default setting and improved checking in hasAccess()

### PR DESCRIPTION
**Pull request by MartijnRas**
I noticed a strange default setting of remoteuser.

I also noticed hasAccess() did not check this default setting.

Furthermore I think the following code in hasAccess() should at least return false, even better would be to remove it completely, as this feels like circumventing all security mechanisms:

        if(trim($conf['remoteuser']) == '') {
            return true;
        }

### CODE REVIEW

**Comment by splitbrain** [comment]
@MartijnRas can you please check the failing test cases?

**Comment by MartijnRas** [comment]
@splitbrain can you please review the updated test as i might have missed some of the intricacies of how the combinations of settings (remote, remoteuser, acl) are supposed to work?

**Comment by splitbrain** [comment]
@dom-mel could you have a look at this?

**Comment by Klap-in** [comment]
Is this security related and useful for comimg release? or is it fine, but just an improvement?

File: inc/remote.php [hunk 1]
```diff
@@ -175,6 +175,9 @@ public function hasAccess() {
         if (!$conf['remote']) {
             return false;
         }
+        if(trim($conf['remoteuser']) == '!!not set!!') {
```
Comment (Chris--S) [inline inc/remote.php]: There is a dependency here that at least warrants a comment.

**Comment by dom-mel** [comment]
looks good to me so :+1:

**Pull request merged**